pkg/chart/chart.go

```diff
@@ -145,14 +157,26 @@ func readResponseBody(res *http.Response) ([]byte, error) {
 	return body, nil
 }
 
-func parseIndex(data []byte) (*repo.IndexFile, error) {
-	index := &repo.IndexFile{}
-	err := yaml.Unmarshal(data, index)
-	if err != nil {
-		return index, err
+func checksum(data []byte) string {
+	hasher := sha1.New()
+	hasher.Write(data)
+	return string(hasher.Sum(nil))
+}
+
+func parseIndex(repoURL string, data []byte) (*repo.IndexFile, error) {
+	sha := checksum(data)
```

Comment:
I'm not sure the cache check should be part of `parseIndex`, it would make more sense if we checked the cache in a separate function before `parseIndex`.